meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch

```diff
@@ -0,0 +1,37 @@
+From 86e59304a4e110440286aab9f5550114f3985eab Mon Sep 17 00:00:00 2001
+From: Jun Nie <jun.nie@linaro.org>
+Date: Mon, 8 Jul 2019 15:28:25 +0800
+Subject: [PATCH 2/2] core: clear original dtb data in special case
+
+clear original dtb data if CFG_DT_ADDR is specified and
+CFG_DT_OVERLAY is not enabled. Because we do not need any
+dtb data from previous stage in this case.
+
+Otherwise there may be residue dtb data in DRAM after reboot.
+Bug happens after many times reboot in u-boot because
+dtb overlay data is added to residue dtb data for many times
+and memory is exhausted by dtb overlay data.
+
+Signed-off-by: Jun Nie <jun.nie@linaro.org>
+---
+ core/arch/arm/kernel/generic_entry_a64.S | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/core/arch/arm/kernel/generic_entry_a64.S b/core/arch/arm/kernel/generic_entry_a64.S
+index 54cc25a..9835c1a 100644
+--- a/core/arch/arm/kernel/generic_entry_a64.S
++++ b/core/arch/arm/kernel/generic_entry_a64.S
+@@ -56,6 +56,10 @@ FUNC _start , :
+ 	mov	x19, x0		/* Save pagable part address */
+ #if defined(CFG_DT_ADDR)
+ 	ldr     x20, =CFG_DT_ADDR
++#ifndef CFG_DT_OVERLAY
++	mov	x14, #0x0       /* clear dtb magic value */
++	str	x14, [x20]
++#endif
```

Comment:
This confuses me.
Aren't we using DTB overlay on i.MX8 ?

Reply:
We are using DTB overlay with CFG_EXTERNAL_DTB_OVERLAY, but not CFG_DT_OVERLAY.

Reply:
I also got questions to this. As you modified a64 code. But we also got the same problems on arm-32 platforms. How do we address it?

Reply:
Because there is no suitable configuration to limit the code and the logic is politics as you mention. Also because this PR is a fix to corner case of test system, so I assume this is a temp solution to imx8mm.
Anyway, I already update patch with change in init_overlay code, although the condition on CFG_DT_ADDR still breaks current optee document.

Reply:
Lets just put the patch into our Linaro tree.

Reply:
That way we can go back and look at this again if we have to, or propose it for upstream.

Reply:
ok. Agree. Let's just keep this patch locally.